xgcm/grid.py

```diff
@@ -1255,45 +1258,42 @@ def check_neighbor(link, position):
             self.axes[axis]._facedim = facedim
             self.axes[axis]._connections = axis_links
 
-    _metrics = {}
-    
     @property
     def metrics(self):
         return self._metrics
 
     @metrics.setter
     def metrics(self, key, value):
-        
-        for key, metric_vars in key, value:
-            metric_axes = frozenset(_maybe_promote_str_to_list(key))
-            if not all([ma in self.axes for ma in metric_axes]):
-                raise KeyError(
-                    "Metric axes %r not compatible with grid axes %r"
-                    % (metric_axes, tuple(self.axes))
-                )
-            # initialize empty list
-            self._metrics[metric_axes] = []
-            for metric_varname in _maybe_promote_str_to_list(metric_vars):
-                if metric_varname not in self._ds:
-                    raise KeyError(
-                        "Metric variable %s not found in dataset." % metric_varname
-                    )
-                # resetting coords avoids potential broadcasting / alignment issues
-                metric_var = self._ds[metric_varname].reset_coords(drop=True)
-                # TODO: check for consistency of metric_var dims with axis dims
-                # check for duplicate dimensions among each axis metric
-                self._metrics[metric_axes].append(metric_var)
-        
-        self._metrics[key] = value
-    
-    def _assign_metrics(self, metrics):
-        """
-        metrics should look like
-           {('X', 'Y'): ['rAC']}
-        check to make sure everything is a valid dimension
-        """
 
-        self._metrics = {}
+        metric_axes = frozenset(_maybe_promote_str_to_list(key))
+        if not all([ma in self.axes for ma in metric_axes]):
+            raise KeyError(
+                "Metric axes %r not compatible with grid axes %r"
+                % (metric_axes, tuple(self.axes))
+            )
```

Comment:
You could us an f-string here:

```suggestion
            raise KeyError(
                f"Metric axes {metric_axes!r} not compatible with grid axes {tuple(self.axes)!r}"
            )
```